website/docs/file_sizing.md

```diff
@@ -3,51 +3,76 @@ title: "File Sizing"
+
+- In case 2: If the Hudi table has a file group with a Parquet base file and an associated log file from updates, or this file group is under a requested or inflight compaction, no more inserts can go into this file group to automatically size the Parquet file. Only after the compaction has been performed, and there are NO log files associated with the base Parquet file, can new inserts be sent to auto-size that parquet file.
+
+Here are the essential configurations:
+
+| Parameter Name | Default  | Description | Scope | Since Version                          |
+|----------------|--------|----------|---------------|--------------------------------------|
+| hoodie.parquet.small.file.limit | 104857600 (100MB) | During an insert and upsert operation, we opportunistically expand existing small files on storage instead of writing new files to keep the number of files optimum. This config sets the file size limit below which a storage file becomes a candidate to be selected as such a `small file`. By default, treat any file <= 100MB as a small file. Also note that if this is set to <= 0, Hudi will not try to get small files and directly write new files. | Write COW, MOR | 0.4.0 |
+| hoodie.parquet.max.file.size |125829120 (120MB) | This config is the target size in bytes for parquet files produced by the Hudi write phases. For DFS, this needs to be aligned with the underlying filesystem block size for optimal performance.  | Write COW, MOR  | 0.4.0 | 
+| hoodie.logfile.max.size | 1073741824 (1GB) | This is the log file max size in bytes. This is the maximum size allowed for a log file before it is rolled over to the next version. | Write MOR  | 0.4.0 | 
+| hoodie.merge.small.file.group.candidates.limit | 1 | This limits the number of file groups, whose base file satisfies the small-file limit to be considered for appending records during an upsert operation. This is only applicable for MOR tables. | Write MOR | 0.4.0 |
+
+
+## Auto-Sizing With Clustering​
+[Clustering](https://hudi.apache.org/docs/next/clustering) is a service that allows you to combine small files into larger ones while at the same time (optionally) changing the data layout by sorting or applying space-filling curves like Z-order or Hilbert curve. We won’t go into all the details about clustering here, but please refer to the [clustering section](https://hudi.apache.org/docs/next/clustering) for more details. 
+
+Clustering is very handy for file sizing so you can have faster queries. When you ingest data, you may still have a lot of small files (depending on your configurations and the data size from ingestion i.e., input batch). In this case, you will want to cluster all the small files to larger files in one write operation to improve query performance. Setting configs for this use case is unnecessary, other than running a job (i.e., Spark job). Optionally, you can customize the file sizing using the configs down below. 
```

Comment:
Please clarify what you mean by - "Setting configs for this use case is unnecessary, other than running a job (i.e., Spark job)." . this is not clear.